anthropic-java-core/src/main/kotlin/com/anthropic/backends/AnthropicBackend.kt

```diff
@@ -0,0 +1,123 @@
+    @get:JvmName("baseUrl") val baseUrl: String,
+) : Backend {
+
+    companion object {
+        private const val PRODUCTION_URL = "https://api.anthropic.com"
+        private const val ANTHROPIC_VERSION = "2023-06-01"
+        private const val ENV_API_KEY = "ANTHROPIC_API_KEY"
+        private const val ENV_AUTH_TOKEN = "ANTHROPIC_AUTH_TOKEN"
+        private const val HEADER_API_KEY = "X-Api-Key"
+        private const val HEADER_AUTHORIZATION = "Authorization"
+        private const val HEADER_VERSION = "anthropic-version"
+
+        @JvmStatic fun builder() = Builder()
+
+        @JvmStatic fun fromEnv(): AnthropicBackend = builder().fromEnv().build()
+    }
+
+    override fun serviceEndpoint(): String = baseUrl
+
+    override fun prepareRequest(request: HttpRequest): HttpRequest {
+        if (request.headers.names().contains(HEADER_VERSION)) {
+            throw AnthropicInvalidDataException(
+                "Request already prepared for Anthropic.")
+        }
+
+        return request.toBuilder()
+            .putHeader(HEADER_VERSION, ANTHROPIC_VERSION)
+            .build()
+    }
+
+    override fun authorizeRequest(request: HttpRequest): HttpRequest {
+        if (request.headers.names().contains(HEADER_API_KEY )
+                || request.headers.names().contains(HEADER_AUTHORIZATION)) {
+            throw AnthropicInvalidDataException(
+                "Request already authorized for Anthropic.")
+        }
+
+        return request.toBuilder()
+            .apply {
+                // Only one of these two properties will be set, so only one
+                // header will be added.
+                apiKey?.let { putHeader(HEADER_API_KEY, it) }
+                authToken?.let { putHeader(HEADER_AUTHORIZATION, "Bearer $it") }
+            }.build()
+    }
+
+    override fun close() {}
+
+    /**
+     * A builder for a [AnthropicBackend] used to connect an Anthropic client to
+     * an Anthropic backend service.
+     *
+     * The authorization credentials can be extracted from the environment and
+     * set on the builder by calling [fromEnv] before calling [build] to create
+     * the [AnthropicBackend]. Alternatively, set the credentials explicitly via
+     * [apiKey] or [authToken] before calling [build].
+     */
+    class Builder internal constructor() {
+        private var apiKey: String? = null
+        private var authToken: String? = null
+        private var baseUrl: String = PRODUCTION_URL
+
+        fun fromEnv() = apply {
+            apiKey = System.getenv(ENV_API_KEY)
+            authToken = System.getenv(ENV_AUTH_TOKEN)
+
+            if (apiKey == null && authToken == null) {
+                throw AnthropicException(
+                    "No API key set in $ENV_API_KEY environment variable and " +
+                            "no authorization token set in $ENV_AUTH_TOKEN " +
+                            "environment variable. Please set exactly one.")
+            }
```

Comment:
I don't think we ever enforced this before. Why enforce it now?

Reply:
Same as https://github.com/anthropics/anthropic-sdk-java/pull/111#discussion_r1972134516

I don't want to add new constraints in this PR that are unrelated to Bedrock and Vertex.